src/ngraph_conversions.h

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#include "ngraph_builder.h"
+#include "ngraph_log.h"
+
+namespace ngraph_bridge {
+
+template<size_t a, size_t b, size_t c, size_t d>
+void Reshape(std::shared_ptr<ng::Node>& ng_node) {
+  static_assert(a < 4 && b < 4 && c < 4 && d < 4,
+                "Number of dimensions cannot exceed 4");
+  static_assert(a != b && a != c && a != d && b !=c && b != d && c != d,
+                "Dimensions indices cannot be equal");
+  auto& s = ng_node->get_shape();
+  ng::Shape reshaped_shape{s[a], s[b], s[c], s[d]};
+  NGRAPH_VLOG(3) << "reshaped_shape: " << ng::join(reshaped_shape);
+  ng_node = make_shared<ng::op::Reshape>(
+      ng_node, ng::AxisVector{a, b, c, d}, reshaped_shape);
+}
+
+void NhwcToNgraph() {}
+
+template <typename... Arguments, typename T>
+void NhwcToNgraph(const std::vector<T>& source, std::vector<size_t>& dest,
+                  Arguments&&... remaining) {
+  dest[0] = source[1];
+  dest[1] = source[2];
+  NhwcToNgraph(remaining...);
+}
+template <typename... Arguments>
+void NhwcToNgraph(std::shared_ptr<ng::Node>& ng_node,
+                         Arguments&&... remaining) {
+  Reshape<0, 3, 1, 2>(ng_node);
+  NhwcToNgraph(remaining...);
+}
+
+void NchwToNgraph() {}
+
+template <typename... Arguments, typename T>
+void NchwToNgraph(const std::vector<T>& source, std::vector<size_t>& dest,
+                  Arguments&&... remaining) {
+  dest[0] = source[2];
+  dest[1] = source[3];
+  NchwToNgraph(remaining...);
+}
+
+template <typename... Arguments>
+void TensorflowToNgraph(bool is_nhwc, std::shared_ptr<ng::Node>& ng_input,
```

Comment:
So I guess this function is doing two things:

(1) reshaping the node, which may be in NHWC, to NCHW
(2) extracting the "2D" nGraph-flavored parameters used by batched ops from the "4D" TensorFlow-flavored parameters (where "parameters" means strides, dilation, pooling window shape, ...).

The variadic template stuff is cool but I think ultimately this will be easier to read if we do something like this:

```
ng_input = BatchToNGraph(is_nhwc, ng_input);
ng_strides = BatchedOpParamToNGraph(is_nhwc, tf_strides);
ng_dilations = BatchedOpParamToNGraph(is_nhwc, tf_dilations);
...
```

Reply:
Yup.

It's quite a bit more typing though, no?  I'm not sure how it's easier to read either.

Reply:
Simple statements are easy to understand and maintain. Please use the suggested functions by Adam.

Reply:
@aprocter is there a consensus on this? I updated it to use `std::pair` for clarity.  I'm willing to change to avijit's suggestion but ideally this would be the last rewrite.